src/main/java/org/testng/reporters/FailedReporter.java

```diff
@@ -143,25 +110,31 @@ private void generateXmlTest(ISuite suite,
       // sorted, we don't need to sort them again.
       //
       List<ITestNGMethod> result = Lists.newArrayList();
+      Set<ITestNGMethod> relevantConfigs = Sets.newHashSet();
       for (ITestNGMethod m : context.getAllTestMethods()) {
         if (methodsToReRun.contains(m)) {
           result.add(m);
+          relevantConfigs.addAll(getAllApplicableConfigs(m.getTestClass()));
         }
       }
-
-      methodsToReRun.clear();
-      Collection<ITestNGMethod> invoked= suite.getInvokedMethods();
-      for(ITestNGMethod tm: invoked) {
-        if(!tm.isTest()) {
-          methodsToReRun.add(tm);
-        }
-      }
-
-      result.addAll(methodsToReRun);
+      result.addAll(relevantConfigs);
       createXmlTest(context, result, xmlTest);
     }
   }
 
+  private Set<ITestNGMethod> getAllApplicableConfigs(ITestClass iTestClass) {
```

Comment:
Fixed the static part.

> If you pass the set as param, you won't have to create a new one each time.

I didn't quite understand this. Since I am extracting all the configuration methods for every `org.testng.ITestClass` I chose to create a new one each time.